spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/store/JdbcChannelMessageStore.java

```diff
@@ -392,6 +394,9 @@ public void afterPropertiesSet() throws Exception {
 			logger.warn("The jdbcTemplate's fetchsize is not 1. This may cause FIFO issues with Oracle databases.");
 		}
 
+		if (this.beanFactory != null) {
+			this.messageBuilderFactory = IntegrationUtils.getMessageBuilderFactory(this.beanFactory);
```

Comment:
To be honest: I followed **exactly** that rule. If there is an `InitializingBean`, I did that in the `afterPropertiesSet()`, otherwise lazy `getMessageBuilderFactory()`.
Have I missed anything ?